libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -319,18 +337,27 @@ void AC_PrecLand::check_target_status(float rangefinder_alt_m, bool rangefinder_
 
     if (_current_target_state == TargetState::TARGET_RECENTLY_LOST) {
         // check if it's nearby/found recently, else the status will be demoted to "TARGET_LOST"
-        Vector2f curr_pos;
-        if (AP::ahrs().get_relative_position_NE_origin(curr_pos)) {
-            const float dist_to_last_target_loc_xy = (curr_pos - Vector2f{_last_target_pos_rel_origin_NED.x, _last_target_pos_rel_origin_NED.y}).length();
-            const float dist_to_last_loc_xy = (curr_pos - Vector2f{_last_vehicle_pos_NED.x, _last_vehicle_pos_NED.y}).length();
+        Vector3f curr_pos;
+        if (AP::ahrs().get_relative_position_NED_origin(curr_pos)) {
+            const Vector3f _last_target_pos_rel_vehicle_NED = _last_target_pos_rel_origin_NED - curr_pos;
+            // sqaured distance towards last target position along direction of approach
+            const float dist_to_last_target_pos_along_doa_sq = _last_target_pos_rel_vehicle_NED.projected(_direction_of_approach).length_squared();
+            // horizontal distance (i.e., perpendicular to the direction of approach) towards last target position
+            const float dist_to_last_target_pos_hor = safe_sqrt(MAX(0, _last_target_pos_rel_vehicle_NED.length_squared() - dist_to_last_target_pos_along_doa_sq));
+
+            const Vector3f _del_vehicle_pos = curr_pos - _last_vehicle_pos_NED;
+            // distance between current vehicle position and last vehicle position along the direction of approach
+            const float dist_to_last_pos_along_doa_sq = _del_vehicle_pos.projected(_direction_of_approach).length_squared();
+            // horizontal distance (i.e., perpendicular to the direction of approach) between current and last vehicle position
+            const float dist_to_last_pos_hor = safe_sqrt(MAX(0, _del_vehicle_pos.length_squared() - dist_to_last_pos_along_doa_sq));
```

Comment:
You can avoid this sqrt by comparing against `sq(LANDING_TARGET_LOST_DIST_THRESH_M)`

Reply:
Done! Thanks for the suggestion :)